src/source/source_cache.js

```diff
@@ -102,6 +102,12 @@ class SourceCache extends Evented {
         }
     }
 
+    setData(data: GeoJSON | string) {
+        if (this._source && this._source.setData) {
+            this._source.setData(data);
+        }
+    }
```

Comment:
Instead of adding this geojson-source-specific method to `SourceCache`, we can use `sourceCache.getSource().setData(...)` in the `Style` method handling this diff operation.